core/libp2p/transport/connection.hpp

```diff
@@ -0,0 +1,39 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_CONNECTION_HPP
+#define KAGOME_CONNECTION_HPP
+
+#include <optional>
+#include <vector>
+
+#include "common/result.hpp"
+#include "libp2p/basic/closeable.hpp"
+#include "libp2p/basic/readable.hpp"
+#include "libp2p/basic/writable.hpp"
+#include "libp2p/common/network_message.hpp"
+#include "libp2p/common/peer_info.hpp"
+#include "libp2p/multi/multiaddress.hpp"
+
+namespace libp2p::transport {
+  /**
+   * Point-to-point link to the other peer
+   */
+  class Connection : public basic::Readable,
+                     public basic::Writable,
+                     public basic::Closeable {
```

Comment:
Writable and closeable must be protected inherited. For now it is enough to leave to do to make them protected in future